pinot-core/src/main/java/org/apache/pinot/core/segment/processing/mapper/SegmentMapper.java

```diff
@@ -62,29 +63,30 @@
  */
 public class SegmentMapper {
   private static final Logger LOGGER = LoggerFactory.getLogger(SegmentMapper.class);
-
-  private List<RecordReaderFileConfig> _recordReaderFileConfigs;
-  private List<RecordTransformer> _customRecordTransformers;
   private final SegmentProcessorConfig _processorConfig;
   private final File _mapperOutputDir;
-
   private final List<FieldSpec> _fieldSpecs;
   private final boolean _includeNullFields;
   private final int _numSortFields;
-
   private final CompositeTransformer _recordTransformer;
   private final TimeHandler _timeHandler;
   private final Partitioner[] _partitioners;
   private final String[] _partitionsBuffer;
   // NOTE: Use TreeMap so that the order is deterministic
   private final Map<String, GenericRowFileManager> _partitionToFileManagerMap = new TreeMap<>();
+  private AdaptiveSizeBasedWriter _adaptiveSizeBasedWriter;
+  private List<RecordReaderFileConfig> _recordReaderFileConfigs;
+  private List<RecordTransformer> _customRecordTransformers;
+  private final int _totalNumRecordReaders;
```

Comment:
This looks a a little weird :). Can we consider the following, to keep this logic simple?

We can add 2 types of logs
1) SegmentProcessFrameWork can log how many readers it passes to the mapper for every iteration.
2) Mapper shows how many readers it completed from the total it got from SegmentProcessFrameWork and where it paused. Basically we don't change any of the mapper code for logging. 

Mainly we want to know how mappers are progressing and whether its stuck.

Reply:
Changed up the logging keeping in mind your suggestions. In addition to the points, I have also added a log in `SegmentProcessorFramework` on the progress of RecordReaders globally.